sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -824,9 +827,20 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
   }
 
   private buildQueueResponse(response: HttpOperationResponse): QueueResponse {
-    const queue = buildQueue(response.parsedBody);
-    const queueResponse: QueueResponse = Object.assign(queue || {}, { _response: response });
-    return queueResponse;
+    try {
+      const queue = buildQueue(response.parsedBody);
+      const queueResponse: QueueResponse = Object.assign(queue || {}, { _response: response });
+      return queueResponse;
+    } catch (err) {
```

Comment:
Don't we have to add a similar try/catch to `buildListQueuesResponse()` as well? Because we can `buildQueue()` there which can throw errors when unexpected output is returned from the service.

Same for topics, subscriptions and rules.

Reply:
the `buildListX` operations use `buildX` to construct the response.
There doesn't seem to be a case that would fail from code outside of the `buildX` steps

Reply:
I am referring to the case where `buildX` throws errors. Since they are not being caught in `buildListXResponse`, they will bubble up to the user. Don't we want to catch them and throw a RestError just like we do in `buildXResponse`

Reply:
The error thrown from `buildX` is a `RestError` that will be bubbling up all the way to the user since nothing else is catching it in between right?

Reply:
Oh oops, I meant to add the error handling to `buildX` and not `buildXResponse` since the former is used within `buildXResponse` and as well as `buildListX`

Reply:
Updated to add error handling around `buildXResponse` and `buildListXResponse` so as to factor in response body type handling for arrays as well. This is just to safeguard for code changes to internal parsing utilities, should they be updated in future.